google/resource_google_project.go

```diff
@@ -173,6 +177,35 @@ func resourceGoogleProjectCreate(d *schema.ResourceData, meta interface{}) error
 	return nil
 }
 
+func resourceGoogleProjectCheckPreRequisites(config *Config, d *schema.ResourceData) error {
+	ib, ok := d.GetOk("billing_account")
+	if !ok {
+		return nil
+	}
+	ba := "billingAccounts/" + ib.(string)
+	const perm = "billing.resourceAssociations.create"
+	req := &cloudbilling.TestIamPermissionsRequest{
+		Permissions: []string{perm},
+	}
+	resp, err := config.clientBilling.BillingAccounts.TestIamPermissions(ba, req).Do()
+	if err != nil {
+		return fmt.Errorf("failed to check permissions on billing account %q: %v", ba, err)
+	}
+	if has := sliceContains(perm, resp.Permissions); !has {
```

Comment:
nit: we already have a function in this package that does this, so you can use stringInSlice(resp.Permissions, perm). You also don't use the value of `has` anywhere so you could just do:
```
if !stringInSlice(resp.Permissions, perm) {
...
}
```

Reply:
Done.